ops/terraform/services/migrator/deploy.groovy

```diff
@@ -138,4 +156,32 @@ boolean canMigratorDeploymentProceed(String sqsQueueName, String awsRegion) {
     }
 }
 
+// checks to determine whether the migrator deployment is required
+// returns true when the latest versioned migration script is newer
+// than the database's stored schema version
+// otherwise false
+boolean isMigratorDeploymentRequired(String bfdEnv, String awsRegion) {
+    println "Comparing schema migration versions..."
+    // Initialize stored schema version
+    int storedSchemaVersion = 0
+    // check SSM Parameter Store
+    try {
+        storedSchemaVersion = awsSsm.getParameter(
+                parameterName: "/bfd/${bfdEnv}/common/nonsensitive/database_schema_version",
+                awsRegion: awsRegion
+        ) as Integer
+    } catch(Exception ex) {
+        echo "Exception has been encountered getting the storedSchemaVersion from aws ssm, missing ssm parameter for stored schema version."
+        return true
+    }
+    echo "Stored Schema Version : ${storedSchemaVersion}"
+
+    // check latest available versioned migration
+    latestAvailableMigrationVersion = sh(returnStdout: true, script: "./ops/jenkins/scripts/getLatestSchemaMigrationScriptVersion.sh") as Integer
+    echo "Latest Available Migration Version: ${latestAvailableMigrationVersion}"
+
+    // compare and determine
+    return latestAvailableMigrationVersion > storedSchemaVersion
```

Comment:
What's the format of these version numbers? Just wanna make sure that this will behave as expected if the versions are strings and something like "1.0.5" or "2.0"

Reply:
whole numbers only

Reply:
> What's the format of these version numbers? Just wanna make sure that this will behave as expected if the versions are strings and something like "1.0.5" or "2.0"

Currently, the schema migration scripts are versioned with simple, strictly incrementing integers. This simple casting of strings to integers paired with integer comparison seems to be sufficient for our current use.